### DISCUSSION TITLE
@RunOnVertxContext annotation

### DISCUSSION THREAD

**Discussion by Angel0926** [quarkusio/quarkus#35151] [Q&A]
i annotated my repository class with them :(service tests work with annotation @RunOnVertxContext in class level)
@QuarkusTest
@WithSession
@RunOnVertxContext. 
class ABC. 

created tests like this  
 @Test
    void getmethod(UniAsserter uniAsserter){
uniAsserter.assertThat(....)
}

when i run tests from class level  i got  "java.lang.IllegalStateException: Can't get the context safety flag: the current context is not a duplicated context" why?

in same time if i anotate every method with @RunOnVertxContext and run, my tests pass.

How can i fix this issue?

**Comment by Angel0926** [comment]
@maxandersen  @geoand

**Comment by geoand** [comment]
`@RunOnVertxContext` is only for test methods

**Reply by Angel0926** [reply]
yes i knew i put this annotation in service test class and it works but it doesn't work in repository test class but want to put annotation in each method to work

**Reply by geoand** [reply]
Please update (and format) the code so I can actually see what you have done - what works and what doesn't

**Reply by Angel0926** [reply]
@ApplicationScoped
@WithSession
public class Repo{}

@ApplicationScoped
public class ServiceImpl{}




i annotated my repository's class with annotation @RunOnVertxContext in class level
@QuarkusTest
@WithSession
@RunOnVertxContext.
class RepositoryTest{

@Test
void myMethod(UniAsserter uni asserter{
}

}
in this case my method does not work and i got "java.lang.IllegalStateException: Can't get the context safety flag: the current context is not a duplicated context" exception, and i need to put annotation in each method to work. But when i try same   in my service class it is work, for example:

@QuarkusTest
@WithSession
@RunOnVertxContext.
class ServiceImplTest{
@Test
void myMethod(UniAsserter uni asserter{
}

}

it is works

**Reply by Angel0926** [reply]
?
